rust/processor/src/grpc_stream.rs

```diff
@@ -431,12 +430,13 @@ pub async fn create_fetcher_loop(
                     for txns in pb_txn_chunks {
                         let size_in_bytes = average_size_in_bytes * txns.len() as u64;
                         let txn_pb = TransactionsPBResponse {
-                            transactions: r.transactions,
+                            transactions: txns,
                             chain_id,
                             start_version,
                             end_version,
-                            start_txn_timestamp,
-                            end_txn_timestamp,
+                            // TODO: this is only for gap checker + filtered txns, but this is wrong
```

Comment:
thoughts on sending the ProcessingResult straight to the gap detector without having to wrap it up in the struct and send to the processor tasks? that way we don't have to have hacks for the filtered txn's in the processor tasks.

Reply:
yeah, I think the overall approach of writing from multiple places + counter is good; could we save that for a further PR? I think we could have an enum that's like `completed(start_version, end_version)` `multi_version(start_version, end_version, num_queries)` etc and use that for communication instead of `ProcessingResult`